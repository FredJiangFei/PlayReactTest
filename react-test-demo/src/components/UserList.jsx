import React from 'react';
import { Link } from 'react-router-dom';

function UserList({ users }) {
  return users.length ? (
    <ul>
      {users.map((x) => (
        <li key={x.id}>
          <Link to={{ pathname:"/userDetails", state: { userId: x.id }}}>
            {`${x.first_name} ${x.last_name}`}
          </Link>
        </li>
      ))}
    </ul>
  ) : (
    'no users'
  );
}

export default UserList;
