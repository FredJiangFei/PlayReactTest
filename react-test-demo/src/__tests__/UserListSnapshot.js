import { render } from '@testing-library/react';
import React from 'react';
import UserList from '../components/UserList';

test('renders the users snapshot', () => {
  // act
  var users = [
    {
      id: 7,
      email: 'michael.lawson@reqres.in',
      first_name: 'Michael',
      last_name: 'Lawson'
    },
    {
      id: 8,
      email: 'lindsay.ferguson@reqres.in',
      first_name: 'Lindsay',
      last_name: 'Ferguson'
    }
  ];
  const {container} = render(<UserList users={users} />)

  // assert
  expect(container.firstChild).toMatchSnapshot()
});