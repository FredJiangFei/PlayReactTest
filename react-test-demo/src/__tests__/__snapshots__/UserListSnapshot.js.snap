// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the users snapshot 1`] = `
<ul>
  <li>
    Michael Lawson
  </li>
  <li>
    Lindsay Ferguson
  </li>
</ul>
`;
